src/pages/ReportAddApproverPage.tsx

```diff
@@ -0,0 +1,205 @@
+import {addReportApprover} from '@libs/actions/IOU';
+import {canUseTouchScreen} from '@libs/DeviceCapabilities';
+import Navigation from '@libs/Navigation/Navigation';
+import type {PlatformStackScreenProps} from '@libs/Navigation/PlatformStackNavigation/types';
+import type {ReportChangeApproverParamList} from '@libs/Navigation/types';
+import {getSearchValueForPhoneOrEmail, sortAlphabetically} from '@libs/OptionsListUtils';
+import {getMemberAccountIDsForWorkspace, isPolicyAdmin} from '@libs/PolicyUtils';
+import {getDisplayNameForParticipant, isMoneyRequestReport, isMoneyRequestReportPendingDeletion} from '@libs/ReportUtils';
+import tokenizedSearch from '@libs/tokenizedSearch';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import NotFoundPage from './ErrorPage/NotFoundPage';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
+import withReportOrNotFound from './home/report/withReportOrNotFound';
+
+type ReportAddApproverPageProps = WithReportOrNotFoundProps & PlatformStackScreenProps<ReportChangeApproverParamList, typeof SCREENS.REPORT_CHANGE_APPROVER.ADD_APPROVER>;
+
+function ReportAddApproverPage({report, isLoadingReportData, policy}: ReportAddApproverPageProps) {
+    const styles = useThemeStyles();
+    const {translate, localeCompare} = useLocalize();
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const [selectedApproverEmail, setSelectedApproverEmail] = useState<string | undefined>(undefined);
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST, {canBeMissing: false});
+    const [countryCode] = useOnyx(ONYXKEYS.COUNTRY_CODE, {canBeMissing: false});
+
+    const currentUserDetails = useCurrentUserPersonalDetails();
+
+    const [allApprovers, setAllApprovers] = useState<SelectionListApprover[]>([]);
+    const shouldShowTextInput = allApprovers?.length >= CONST.STANDARD_LIST_ITEM_LIMIT;
+
+    const employeeList = policy?.employeeList;
+    const sections = useMemo(() => {
+        const approvers: SelectionListApprover[] = [];
+
+        if (employeeList) {
+            const policyMemberEmailsToAccountIDs = getMemberAccountIDsForWorkspace(employeeList);
+            const availableApprovers = Object.values(employeeList)
+                .map((employee): SelectionListApprover | null => {
+                    const isAdmin = employee?.role === CONST.REPORT.ROLE.ADMIN;
+                    const email = employee.email;
+
+                    if (!email) {
+                        return null;
+                    }
+                    const accountID = Number(policyMemberEmailsToAccountIDs[email] ?? CONST.DEFAULT_NUMBER_ID);
+
+                    if (report.managerID === accountID) {
```

Comment:
NAB could add some comment here about why we are filtering out the current manager e.g. they are already assigned as manager. Maybe it is obvious.